lua/telescope/builtin/lsp.lua

```diff
@@ -145,18 +142,14 @@ lsp.document_symbols = function(opts)
 end
 
 lsp.code_actions = function(opts)
-  local params = vim.F.if_nil(opts.params, vim.lsp.util.make_range_params())
+  local params = vim.F.if_nil(opts.params, vim.lsp.util.make_range_params(opts.winnr))
+  local lnum = vim.api.nvim_win_get_cursor(opts.winnr)[1]
 
   params.context = {
-    diagnostics = vim.lsp.diagnostic.get_line_diagnostics(),
+    diagnostics = vim.lsp.diagnostic.get_line_diagnostics(opts.bufnr, lnum - 1),
```

Comment:
?

we should move to `vim.diagnostic.get` here

Reply:
![image](https://user-images.githubusercontent.com/15233006/151678319-1106ccb4-b237-40f5-92d2-62aab25fe6d5.png)

I dont know whats happening here. Can we use `vim.diagnostic.get`?

I think this should do the trick
```
vim.diagnostic.get(opts.bufnr, { lnum = lnum - 1})
```